bucket/bucket_plain.go

```diff
@@ -5,7 +5,49 @@ import (
 )
 
 // MetricOperation is a list of metric operations to use for metric
-type MetricOperation [3]string
+type MetricOperation struct {
+	operations []string
+	Labels     map[string]string
+}
+
+// NewMetricOperation  builds and returns new MetricOperation instance with defined label keys
+func NewMetricOperation(operations ...string) MetricOperation {
+	ops := []string{"-", "-", "-"}
+
+	for i := 0; i < 3; i++ {
+		if len(operations) <= i {
+			break
+		}
+		ops[i] = operations[i]
+	}
+	return MetricOperation{operations: ops}
+}
+
+// WithLabels adds label value to existing MetricOperation instance
+func (m *MetricOperation) WithLabels(labels map[string]string) MetricOperation {
+
+	if m.Labels == nil {
+		m.Labels = labels
+		return *m
+	}
+
+	for k := range m.Labels {
+		m.Labels[k] = ""
+	}
+
+	for k := range labels {
+		if _, ok := m.Labels[k]; !ok {
+			// TODO: handle error properly
+			panic("undefined labelName: " + k)
```

Comment:
Any thoughts how can I raise error here in better go way?

Reply:
either you return the error or panic one, clients will need to recover

Reply:
just ignore it - panicking in stats library is kinda bad idea